modules/server/src/topics/camerapathtopic.cpp

```diff
@@ -0,0 +1,113 @@
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/server/include/topics/camerapathtopic.h>
+
+#include <modules/server/include/connection.h>
+#include <modules/server/servermodule.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/moduleengine.h>
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/navigation/navigationhandler.h>
+#include <openspace/navigation/path.h>
+#include <openspace/navigation/pathnavigator.h>
+#include <ghoul/logging/logmanager.h>
+
+namespace {
+    constexpr std::string_view SubscribeEvent = "start_subscription";
+} // namespace
+
+using nlohmann::json;
+
+namespace openspace {
+
+CameraPathTopic::CameraPathTopic()
+    : _lastUpdateTime(std::chrono::system_clock::now())
+{}
+
+CameraPathTopic::~CameraPathTopic() {
+    if (_dataCallbackHandle != UnsetOnChangeHandle) {
+        ServerModule* module = global::moduleEngine->module<ServerModule>();
+        if (module) {
+            module->removePreSyncCallback(_dataCallbackHandle);
+        }
+    }
+}
+
+bool CameraPathTopic::isDone() const {
+    return _isDone;
+}
+
+void CameraPathTopic::handleJson(const nlohmann::json& json) {
+    std::string event = json.at("event").get<std::string>();
+
+    if (event != SubscribeEvent) {
+        _isDone = true;
+        return;
+    }
+
+    ServerModule* module = global::moduleEngine->module<ServerModule>();
+    _dataCallbackHandle = module->addPreSyncCallback(
+        [this]() {
+            bool isInPath = (global::openSpaceEngine->currentMode()
+                == OpenSpaceEngine::Mode::CameraPath);
+
+            std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+            if (isInPath && (now - _lastUpdateTime) > _cameraPathUpdateTime) {
+                sendCameraPathData();
+                _lastUpdateTime = std::chrono::system_clock::now();
+            }
+        }
+    );
+}
+
+void CameraPathTopic::sendCameraPathData() {
+    const interaction::PathNavigator& pathNavigator =
+        global::navigationHandler->pathNavigator();
+
+    const interaction::Path* path = pathNavigator.currentPath();
+
+    if (!path) {
+        ghoul_assert(path, "Path must exist");
+        return;
+    }
+
+    // The time is not exact, and we only care about the number of seconds. Also,
+    // any negative values should be interpreted as positive
+    int seconds = static_cast<int>(
+        std::ceil(pathNavigator.estimatedRemainingTimeInPath())
```

Comment:
Consider changing this rounding to `round` or `floor`. Now it sometimes seems like the path ends about 1 second before it should. Or actually, maybe don't do any rounding at all and just show the integer version of the number.

(The reason I went with ceil was that with sometimes get values that are slightly smaller than zero, as the estimated time is not 100% correct)